src/Networking/NetworkClient.cpp

```diff
@@ -12,49 +12,27 @@
 static const uint32_t connectTimeout = ConnectTimeout + 1000; // Make sure the socket, if any, is closed before attempting another connection.
 
 NetworkClient::NetworkClient(NetworkResponder *n, NetworkClient *c) noexcept
-	: NetworkResponder(n), interface(nullptr), next(c), whenRequest(0)
+	: NetworkResponder(n), next(c), whenRequest(0)
 {
 	clients = this;
 }
 
-bool NetworkClient::Start(NetworkProtocol protocol, NetworkInterface *iface) noexcept
+bool NetworkClient::Start(NetworkProtocol protocol, NetworkInterface *interface) noexcept
 {
 	if (HandlesProtocol(protocol))
 	{
 		if (skt == nullptr)
 		{
-			if (interface != nullptr)
+			if (millis() - whenRequest < connectTimeout)
```

Comment:
You have removed the check for interface != nullptr here. Are you certain that this function can't be called with interface == nullptr and skt != nullptr? For example, what will happen on a 6HC without a wifi module attached if you try to start MQTT on interface #1 ? Or is it the case the interface == nullptr is actually OK here?

Reply:
The `interface` here was a class member that has been supplanted by keeping track of the interface number instead. 

Furthermore, this check is supplanted by checking in `MqttClient.cpp`. 

Lastly, this code is called only when the specific `NetworkInterface` is in the `NetworkState::connected`. I double checked this, and if the wifi module is not attached, the network state remains at `NetworkState::disabled` and so this code never gets called, even when MQTT is enabled on the wifi interface.

Reply:
Thanks for the clarification..